server/src/main/scala/net/wiringbits/actions/SendNotificationAction.scala

```diff
@@ -0,0 +1,43 @@
+package net.wiringbits.actions
+
+import net.wiringbits.apis.EmailApi
+import net.wiringbits.apis.models.EmailRequest
+import net.wiringbits.config.NotificationsConfig
+import net.wiringbits.repositories.models.UserNotification
+import net.wiringbits.repositories.{UserNotificationsRepository, UsersRepository}
+import net.wiringbits.util.EmailMessage
+
+import java.time.Clock
+import java.time.temporal.ChronoUnit
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class SendNotificationAction @Inject() (
+    emailApi: EmailApi,
+    usersRepository: UsersRepository,
+    userNotificationsRepository: UserNotificationsRepository,
+    notificationsConfig: NotificationsConfig
+)(implicit
+    ec: ExecutionContext,
+    clock: Clock
+) {
+
+  def apply(notification: UserNotification): Future[Unit] = {
+    try {
+      sendEmail(notification)
```

Comment:
This won't work because the method returns a future, this is what you need

```
sendEmail(...).onComplete {
  case Failure(ex) => userNotificationsRepository.setStatusToFailed(...)
  case Success(_) =>()
}
```